src/commands/RedactCommand.ts

```diff
@@ -44,14 +44,22 @@ export async function execRedactCommand(roomId: string, event: any, mjolnir: Mjo
         // Assume it's a permalink
         const parsed = Permalinks.parseUrl(parts[2]);
         const targetRoomId = await mjolnir.client.resolveRoom(parsed.roomIdOrAlias);
+        await mjolnir.protectedRoomsTracker.quarantineMediaForEventId(targetRoomId, parsed.eventId);
         await mjolnir.client.redactEvent(targetRoomId, parsed.eventId);
         await mjolnir.client.unstableApis.addReactionToEvent(roomId, event["event_id"], "✅");
         await mjolnir.client.redactEvent(roomId, processingReactionId, "done processing command");
         return;
     }
 
     const targetRoomIds = targetRoom ? [targetRoom] : mjolnir.protectedRoomsTracker.getProtectedRooms();
+    const medias = mjolnir.protectedRoomsTracker.getMediaIdsForUserIdInRooms(userId, targetRoomIds);
     await redactUserMessagesIn(mjolnir.client, mjolnir.managementRoomOutput, userId, targetRoomIds, false, limit);
+    for (const { domain, mediaId } of medias) {
+        await mjolnir.client.doRequest(
+            "POST",
```

Comment:
for this and elsewhere, do we want to warn on a failed quarantine attempt?

Reply:
We now warn properly.